apps/web-giddh/src/app/shared/primary-sidebar/company-branch/company-branch.component.ts

```diff
@@ -0,0 +1,429 @@
+
+            if (organization && organization.details && organization.details.branchDetails) {
+                this.currentBranchUniqueName = organization.details.branchDetails.uniqueName;
+            }
+        });
+    }
+
+    /**
+     * Releases the occupied memory
+     *
+     * @memberof CompanyBranchComponent
+     */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+    }
+
+    /**
+     * Company filtering method
+     *
+     * @param {*} ev Modal change event
+     * @memberof CompanyBranchComponent
+     */
+    public filterCompanyList(event: any): void {
+        let companies: CompanyResponse[] = [];
+        this.companies$?.pipe(take(1)).subscribe(cmps => companies = cmps);
+
+        this.companyListForFilter = companies?.filter((cmp) => {
+            if (!cmp.alias) {
+                return cmp.name.toLowerCase().includes(event.toLowerCase());
+            } else {
+                return cmp.name.toLowerCase().includes(event.toLowerCase()) || cmp.alias.toLowerCase().includes(event.toLowerCase());
+            }
+        });
+    }
+
+    /**
+     * Change company callback method
+     *
+     * @param {string} selectedCompanyUniqueName Selected company unique name
+     * @param {boolean} [fetchLastState] True, if last state of the company needs to be fetched
+     * @memberof CompanyBranchComponent
+     */
+    public changeCompany(selectedCompanyUniqueName: string, selectBranchUniqueName: string, fetchLastState?: boolean) {
+        this.generalService.companyUniqueName = selectedCompanyUniqueName;
+        const details = {
+            branchDetails: {
+                uniqueName: selectBranchUniqueName
+            }
+        };
+
+        if(selectBranchUniqueName) {
+            this.setOrganizationDetails(OrganizationType.Branch, details);
+        } else {
+            this.setOrganizationDetails(OrganizationType.Company, details);
+        }
+
+        this.store.dispatch(this.loginAction.ChangeCompany(selectedCompanyUniqueName, fetchLastState));
+    }
+
+    /**
+    * Sets the organization details
+    *
+    * @private
+    * @param {OrganizationType} type Type of the organization
+    * @param {OrganizationDetails} branchDetails Branch details of an organization
+    * @memberof CompanyBranchComponent
+    */
+    private setOrganizationDetails(type: OrganizationType, branchDetails: OrganizationDetails): void {
+        const organization: Organization = {
+            type, // Mode to which user is switched to
+            uniqueName: this.activeCompany ? this.activeCompany.uniqueName : '',
+            details: branchDetails
+        };
+        this.store.dispatch(this.companyActions.setCompanyBranch(organization));
+    }
+
+    /**
+     * Refreshes the company list
+     *
+     * @param {Event} event
+     * @memberof CompanyBranchComponent
+     */
+    public refreshCompanies(event: Event): void {
```

Comment:
Please check this method is also present in `primary-sidebar.component.ts`, do remove it from there if unused.